src/services/rfqm_service.ts

```diff
@@ -1112,34 +1112,54 @@ export class RfqmService {
         gasEstimate: number,
     ): Promise<RfqmTransactionSubmissionEntity> {
         const txOptions = {
-            nonce,
+            from: workerAddress,
             gas: gasEstimate,
             gasPrice,
+            nonce,
             value: 0,
         };
 
-        // submit tx to chain
-        // TODO: get transaction hash prior to submission and save to DB first
-        const transactionHash = await this._blockchainUtils.submitCallDataToExchangeProxyAsync(
-            callData,
-            workerAddress,
+        const transactionRequest = await this._blockchainUtils.transformTxDataToTransactionRequestAsync(
             txOptions,
+            callData,
         );
-        logger.info({ orderHash, workerAddress, transactionHash }, 'transaction calldata submitted to exchange proxy');
+        const { signedTransaction, hash } = await this._blockchainUtils.signTransactionAsync(transactionRequest);
 
-        // save tx submission to DB
         const partialEntity: Partial<RfqmTransactionSubmissionEntity> = {
-            transactionHash,
+            transactionHash: hash,
             orderHash,
             createdAt: new Date(),
             from: workerAddress,
             to: this._blockchainUtils.getExchangeProxyAddress(),
             gasPrice,
             nonce,
-            status: RfqmTransactionSubmissionStatus.Submitted,
+            status: RfqmTransactionSubmissionStatus.Presubmit,
         };
 
-        return this._dbUtils.writeRfqmTransactionSubmissionToDbAsync(partialEntity);
+        await this._dbUtils.writeRfqmTransactionSubmissionToDbAsync(partialEntity);
+
+        const transactionHashFromSubmit = await this._blockchainUtils.submitSignedTransactionAsync(signedTransaction);
+
+        if (hash !== transactionHashFromSubmit) {
+            // This should never ever happen
+            logger.error(
+                `Mismatch between transaction hash calculated before submit and after submit:\nBefore submit:\t${hash}\nAfter submit:\t${transactionHashFromSubmit}`,
+            );
```

Comment:
to make this log easier to search upon, can we use the following syntax:
```
logger.error( {
 transactionHashFromSubmit,
 transactionHash: hash
}, "Mismatch between transaction hash calculated before submit and after submit");
```

This will allow us to search on the `transactionHash` and trace it's journey through the log messages

Reply:
Oh good call